### ISSUE TITLE
Allow saving login when I log into a website

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/fenix#5542]
The GV API that triggers doorhangers will be done as part of #6696 

### Why/User Benefit/User Problem
When a user logs in on a website, they can save the login.

### What/Requirements

This is only partially waiting, but since it's one of our MUSTs for the quarter, we can start working on the UI parts.

- [x] [Mock this out so it unblocks subsequent work - will receive API 11/8] Fenix can detect that a login has been submitted on a site - GV bug for autofill API listening for [PasswordManager:onFormSubmit ](https://searchfox.org/mozilla-central/rev/325c1a707819602feff736f129cb36055ba6d94f/toolkit/components/passwordmgr/LoginManagerContent.jsm#1519-1528): https://bugzilla.mozilla.org/show_bug.cgi?id=1577005
- [x] Show a doorhanger w/ save/don't-save options
- [ ] ~~Add Exceptions for sites users don't want to save logins for~~ #7094
- [x] Fenix will write login credentials to Sync Login Storage using existing AS/AC APIs

### Acceptance Criteria (how do I know when I’m done?)
Logins can be saved when submitting on a site
Does not include viewing logins, that is #5544 

#### Not in this bug
Choosing secure local storage, see #5550

**Comment by liuche** [comment]
I chatted w/ GV about this, and at the very highest level, Fenix will be doing login saving (as in, GV will pass the login blob to be saved to Logins storage).

This means that we can start the in-app notification work for saving a login, and just write dummy content to the Logins storage. #5074 should be in Fenix with the next AC version 20.0 on 11/4.

We'll receive a GV API for capture by 11/8, to unblock the other part of this work (triggering the login).

**Comment by liuche** [comment]
Moving this into the sprint because we're moving out #5543 due to new GV work being done on it.

**Comment by ekager** [comment]
Resized to an E3 to build up the UI and saving a "login" to the storage while waiting on the GV APIs.

**Comment by cpeterso** [comment]
The GV bug has been added to GV's November sprint.

https://bugzilla.mozilla.org/show_bug.cgi?id=1577005#c13

**Comment by liuche** [comment]
@ekager could we show this dialog when GV detects that there is a login? Also, can we add the exceptions list? Basically everything except writing the login to the Sync storage.

Would be down to land this preffed off entirely, or Nightly-only just disabling the "Save" button in the dialog.

**Comment by liuche** [comment]
Chatted w/ @AmyYLee and we're going to break out "logins exceptions list" as a separate issue for #7094.

**Comment by sv-ohorvath** [comment]
Issues found on Nightly 1/21:
1. The Save/Update button doesn't dismiss the pop-up but it does add the credentials to the list.
2. On amazon.com, enter an email and password, submit. On the save login popup, the user email is not filled-in.
3. The left side icons are white on 2 devices: LG Nexus 5, HTC Desire 820 ( both Android 6.0.1).

**Comment by ekager** [comment]
1 was filed as #7828 and should be resolved. 
I don't think we need to block this feature on 2/3 and we can file 2 and 3 as new bugs and close this one?

**Comment by sv-ohorvath** [comment]
Created #7886 & https://github.com/mozilla-mobile/fenix/issues/7885. Closing this as fixed on Nightly.
